[WIP] 30th July 2018 - Updated layout for App Lock Screen

vinaysshenoy opened the pull request.

saket commented:
1. The distance between `FORGOT PIN?` button and the card's bottom doesn't match Zeplin
2. The `NOT YOU?` button doesn't match Zeplin

saket commented on app/src/main/java/org/simple/clinic/widgets/Views.kt:
```diff
@@ -109,3 +111,12 @@ fun View.topRelativeTo(superParent: ViewGroup): Int {
+fun View.showToast(msg: CharSequence, longToast: Boolean = false) {
+  val toastLength = if (longToast) Toast.LENGTH_LONG else Toast.LENGTH_SHORT
+  Toast.makeText(context, msg, toastLength).show()
+}
```
Based on the rules used by Jake and other team members of android-ktx, I think this is a wrong usage of extension function. 

https://www.youtube.com/watch?v=st1XVfkDWqk

vinaysshenoy replied:
Got it. The core ktx [package](https://developer.android.com/reference/kotlin/androidx/core/widget/package-summary#extension-functions-summary) has an extension method for toasting via the `Context`. I can either add the library or implement the same signature myself.

Since we already use `work-runtime-ktx`, I guess we can use the core ktx library as well. Thoughts?

saket replied:
Let's add android-ktx

saket commented on app/src/test/java/org/simple/clinic/login/applock/AppLockScreenControllerTest.kt:
```diff
@@ -88,4 +88,18 @@ class AppLockScreenControllerTest {
   }
+
+  @Test
+  fun `when logout is clicked, the ui action should be triggered`() {
```
Hmmm… `the ui action` sounds wrong

vinaysshenoy replied:
Agreed. I just added it in here because I wanted to verify that the ui events are getting triggered. I plan to rename it when we actually implement the feature.

vinaysshenoy commented:
@saket 
- Fixed design inconsistencies
- I added `core-ktx`, but that had deprecated the `toast` methods in the latest version. I decided to directly call `Toast.makeText().show()` in the `AppLockScreen` and did not include `core-ktx`.

saket commented:
Hey @vinaysshenoy I don't see android-ktx in this PR. Is it a WIP?

Btw, I was wondering if we should also add the change-facility button before you move to some other story. When we add a screen for changing facilities in the future, we won't have to update this screen again. Thoughts?

saket commented on app/src/main/java/org/simple/clinic/login/applock/AppLockScreen.kt:
```diff
@@ -103,7 +121,7 @@ class AppLockScreen(context: Context, attrs: AttributeSet) : RelativeLayout(cont
     when (show) {
       true -> errorTextView.visibility = View.VISIBLE
-      else -> errorTextView.visibility = View.GONE
+      else -> errorTextView.visibility = View.INVISIBLE
```
The large gap between the pin text field and forgot-pin button doesn't look very good now that I'm running this new design on my device.

saket commented:
Closing this for now because the design may change.
